packages/insomnia-app/app/ui/components/modals/workspace-duplicate-modal.tsx

```diff
@@ -0,0 +1,157 @@
+import { selectActiveSpace, selectIsLoggedIn, selectSpaces } from '../../redux/selectors';
+import Modal from '../base/modal';
+import ModalBody from '../base/modal-body';
+import ModalFooter from '../base/modal-footer';
+import ModalHeader from '../base/modal-header';
+import { showModal } from '.';
+
+interface Options {
+  workspace: Workspace;
+  apiSpec: ApiSpec;
+  onDone?: () => void;
+}
+
+interface FormFields {
+  newName: string;
+  spaceId: string;
+}
+
+interface InnerProps extends Options, Props {
+  hide: () => void,
+}
+
+const SpaceOption: FC<Space> = space => (
+  <option key={space._id} value={space._id}>
+    {space.name} ({isBaseSpace(space) ? 'Base' : isLocalSpace(space) ? 'Local' : 'Remote'})
+  </option>
+);
+
+const WorkspaceDuplicateModalInternalWithRef: ForwardRefRenderFunction<Modal, InnerProps> = ({ workspace, apiSpec, onDone, hide, vcs }, ref) => {
+  const dispatch = useDispatch();
+
+  const spaces = useSelector(selectSpaces);
+  const activeSpace = useSelector(selectActiveSpace);
+  const isLoggedIn = useSelector(selectIsLoggedIn);
+
+  const title = `Duplicate ${getWorkspaceLabel(workspace).singular}`;
+  const defaultWorkspaceName = getWorkspaceName(workspace, apiSpec);
+
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: {
+      errors,
+    } } = useForm<FormFields>({
+      defaultValues: {
+        newName: defaultWorkspaceName,
+        spaceId: activeSpace._id,
+      },
+    });
+
+  const onSubmit = useCallback(async ({ spaceId, newName }: FormFields) => {
+    const duplicateToSpace = spaces.find(s => s._id === spaceId);
+    if (!duplicateToSpace) {
+      throw new Error('Space could not be found');
+    }
+    
+    const newWorkspace = await workspaceOperations.duplicate(workspace, { name: newName, parentId: spaceId });
+    await models.workspace.ensureChildren(newWorkspace);
+    
+    // Mark for sync if logged in and in the expected space
+    if (isLoggedIn && vcs && isRemoteSpace(duplicateToSpace)) {
+      await initializeLocalProjectAndMarkForSync({ vcs: vcs.newInstance(), workspace: newWorkspace });
+    }
+
+    dispatch(activateWorkspace(newWorkspace));
+    hide();
+    onDone?.();
+  }, [dispatch, hide, isLoggedIn, onDone, spaces, vcs, workspace]);
+
+  return <Modal ref={ref} onShow={reset}>
+    <ModalHeader>{title}</ModalHeader>
+    <ModalBody className="wide">
+      <form className="wide pad" onSubmit={handleSubmit(onSubmit)}>
+        <div className="form-control form-control--wide form-control--outlined">
+          <label>
+            New Name
+            <input {...register('newName', { validate: v => Boolean(v.trim()) || 'Should not be blank' })} />
+            {errors.newName && <div className="font-error space-top">{errors.newName.message}</div>}
+          </label>
+        </div>
+        <div className="form-control form-control--outlined">
+          <label>
+            {strings.space.plural} to duplicate into
```

Comment:
```suggestion
            {strings.space.singular} to duplicate into
```

@wdawson I don't know how much you care about stuff like this so I'm moreso asking to learn your preference, but there's an argument to be made here for this to always be singular because we're always copying into a single space (this isn't a multi-select box at this time)?

<img height="150" src="https://user-images.githubusercontent.com/15232461/127241346-fa2cace2-e689-4ce1-833d-6f73525fd51c.png" />

Reply:
Oops! Yeah this should probably be singular

Reply:
[`3d1a59f` (#3820)](https://github.com/Kong/insomnia/pull/3820/commits/3d1a59fdc5e2edaf47b7af4266d0b24bfbf33e7e)

Reply:
Definitely should be singular! Happy to get the pings for this stuff if there's confusion or contention 👍